### PR TITLE
Configurable log level

### PR DESCRIPTION
**Pull request by wanliAlex**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
configurable log level for `timing log`and `uvicorn log`

* **What is the current behavior?** (You can also link to an open issue here)
not configurable

* **What is the new behavior (if this is a feature change)?**
now it's configurable

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
no

* **Related Python client changes** (link commit/PR here)
no

* **Related documentation changes** (link commit/PR here)
no

* **Other information**:
```bash
docker run --name marqo --privileged -p 8882:8882 --add-host host.docker.internal:host-gateway -e MARQO_LOG_LEVEL=warning marqo_docker_0
```

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Comment by wanliAlex** [comment]
image

Logs adding 1000 documents into index with this PR

```bash
docker run --name marqo --privileged -p 8882:8882 --add-host host.docker.internal:host-gateway -e MARQO_LOG_LEVEL=warning marqo_docker_0
```

**Comment by wanliAlex** [comment]
image

Logs adding 1000 documents into index with this PR

```bash
docker run --name marqo --privileged -p 8882:8882 --add-host host.docker.internal:host-gateway marqo_docker_0
```

File: src/marqo/tensor_search/configs.py [hunk 1]
```diff
@@ -34,6 +34,7 @@ def default_env_vars() -> dict:
         EnvVars.MARQO_MAX_INDEX_FIELDS: None,
         EnvVars.MARQO_MAX_DOC_BYTES: 100000,
         EnvVars.MARQO_MAX_RETRIEVABLE_DOCS: 10000,
-        EnvVars.MARQO_MODELS_TO_PRELOAD: ['hf/all_datasets_v4_MiniLM-L6', "ViT-L/14"]
+        EnvVars.MARQO_MODELS_TO_PRELOAD: ['hf/all_datasets_v4_MiniLM-L6', "ViT-L/14"],
+        EnvVars.MARQO_LOG_LEVEL: "info"
```
Comment (pandu-k) [inline src/marqo/tensor_search/configs.py]: Can you add a comment next to `EnvVars.MARQO_LOG_LEVEL` saying:
`# this env var is set to info by default in run_marqo.sh, which overrides this value `

**Review by tomhamer — APPROVED** [review]

File: src/marqo/tensor_search/tensor_search_logging.py [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
-    logging.basicConfig() 
+    logging.basicConfig()
     logger = logging.getLogger(name)
-    logger.setLevel(logging.INFO)
+
+    log_level = read_env_vars_and_defaults("MARQO_LOG_LEVEL")
+    if log_level == "warning":
+        logger.setLevel(level=logging.WARNING)
+    elif log_level == "info":
+        logger.setLevel(level=logging.INFO)
```
Comment (pandu-k) [inline src/marqo/tensor_search/tensor_search_logging.py:12]: 1. Make sure we mention there are only 2 log levels in the documentation.
2. Can you raise an error if an unsupported log level is provided? In the error mention what log levels are supported, and link to relevant docs.
3. Can you make the log levels case-insensitive?

**Review by pandu-k — CHANGES_REQUESTED** [review]

File: src/marqo/tensor_search/configs.py (hunk 1 above)
Comment (wanliAlex) [reply src/marqo/tensor_search/configs.py]: done

File: src/marqo/tensor_search/tensor_search_logging.py (hunk 2 above)
Comment (wanliAlex) [reply src/marqo/tensor_search/tensor_search_logging.py:12]: done

**Comment by pandu-k** [comment]
DIND OS tests pass: https://github.com/marqo-ai/marqo/actions/runs/4150909911/jobs/7180884258

File: src/marqo/tensor_search/tensor_search_logging.py [hunk 3]
```diff
@@ -1,9 +1,23 @@
...
+    elif log_level == "debug":
+        logger.setLevel(level=logging.DEBUG)
+    elif log_level == "error":
+        logger.setLevel(level=logging.ERROR)
+    else:
+        raise EnvVarError(f"The provided {log_level} is not supported."
+                          f"The {log_level} should be one of `error`, `warning`, `info`, `debug`."
+                          f"Check https://docs.marqo.ai/0.0.13/Advanced-Usage/configuration/ for more info.")
```
Comment (pandu-k) [inline src/marqo/tensor_search/tensor_search_logging.py:20]: Please surround the the user's log level with backticks:
```
`{log_level}`
```
Also, we think about how this looks printed out, it looks odd: 
```
The provided MY_CUSTOM_LEVEL is not supported. The MY_CUSTOM_LEVEL should be one of `error`, `warning`, `info`, `debug`....
```
The user needs to know the problem is with the log level. Can you also make explicit reference to the env var the user needs to correct if they get this error?
```
The provided log level `MY_CUSTOM_LEVEL`, set by the environment variable `MARQO_LOG_LEVEL` is not supported. The log level should be one of `error`, `warning`, `info`, `debug`. 
```

File: src/marqo/tensor_search/tensor_search_logging.py (hunk 2 above)
Comment (pandu-k) [reply src/marqo/tensor_search/tensor_search_logging.py:12]: Thanks Li!

**Review by pandu-k — CHANGES_REQUESTED** [review]

File: src/marqo/tensor_search/tensor_search_logging.py (hunk 3 above)
Comment (wanliAlex) [reply src/marqo/tensor_search/tensor_search_logging.py:20]: thanks! customer first!

**Comment by pandu-k** [comment]
Testing required: 
- DIND OS test against this branch
- Manual testing if log_level = (error, debug) works as expected
- If passing args to uvicorn is also case insensitive

File: src/marqo/tensor_search/tensor_search_logging.py (hunk 3 above)
Comment (wanliAlex) [reply src/marqo/tensor_search/tensor_search_logging.py:20]: done

**Comment by wanliAlex** [comment]
image

```bash
docker run --name marqo --privileged -p 8882:8882 --add-host host.docker.internal:host-gateway -e MARQO_LOG_LEVEL=debug marqo_docker_0
```


Upper case
```bash
docker run --name marqo --privileged -p 8882:8882 --add-host host.docker.internal:host-gateway -e MARQO_LOG_LEVEL=DEBUG marqo_docker_0
```

also passed

**Review by pandu-k — APPROVED** [review]

**Pull request merged**